### PR TITLE
Update gdt and gdtcct podspecs

### PR DESCRIPTION
**Pull request by doudounan**
Update gdt and gdtcct podspecs with frameworks

### CODE REVIEW

File: GoogleDataTransport.podspec [hunk 1]
```diff
@@ -29,6 +29,11 @@ Shared library for iOS SDK data transport needs.
   s.source_files = 'GoogleDataTransport/GDTCORLibrary/**/*'
   s.public_header_files = 'GoogleDataTransport/GDTCORLibrary/Public/*.h'
   s.private_header_files = 'GoogleDataTransport/GDTCORLibrary/Private/*.h'
+  s.frameworks = 'Foundation', 'SystemConfiguration', 'CoreTelephony'
+  s.ios.framework  = 'UIKit'
```
Comment (mikehaney24) [inline GoogleDataTransport.podspec]: I think you can safely omit Foundation, UIKit, AppKit, and WatchKit

File: GoogleDataTransportCCTSupport.podspec [hunk 2]
```diff
@@ -30,7 +30,12 @@ Support library to provide event prioritization and uploading for the GoogleData
 
   s.source_files = 'GoogleDataTransportCCTSupport/GDTCCTLibrary/**/*'
   s.private_header_files = 'GoogleDataTransportCCTSupport/GDTCCTLibrary/Private/*.h'
-
+  s.frameworks = 'Foundation', 'SystemConfiguration', 'CoreTelephony'
```
Comment (mikehaney24) [inline GoogleDataTransportCCTSupport.podspec]: Same as other comment

File: GoogleDataTransport.podspec [hunk 3]
```diff
@@ -29,6 +29,9 @@ Shared library for iOS SDK data transport needs.
   s.source_files = 'GoogleDataTransport/GDTCORLibrary/**/*'
   s.public_header_files = 'GoogleDataTransport/GDTCORLibrary/Public/*.h'
   s.private_header_files = 'GoogleDataTransport/GDTCORLibrary/Private/*.h'
+  s.ios.frameworks = 'SystemConfiguration', 'CoreTelephony'
+  s.osx.frameworks = 'SystemConfiguration', 'CoreTelephony'
+  s.tvos.frameworks = 'SystemConfiguration'
```
Comment (paulb777) [inline GoogleDataTransport.podspec]: Does watchOS need any?

Comment (doudounan) [reply GoogleDataTransport.podspec]: Both two framework is not available in watchOS, and if we update our api to adopt watchOS in the future, I think we will add relative framework for watchOS later.

Comment (paulb777) [reply GoogleDataTransport.podspec]: SG - thanks for checking.

**Pull request merged**